src/griffe/finder.py

```diff
@@ -338,22 +343,21 @@ def _handle_editable_module(path: Path):  # noqa: WPS231
         editable_lines = path.read_text(encoding="utf8").splitlines(keepends=False)
     except FileNotFoundError:
         raise UnhandledEditableModuleError(path)
+
     if path.name.startswith("__editables_"):
         # support for how 'editables' writes these files:
         # example line: F.map_module('griffe', '/media/data/dev/griffe/src/griffe/__init__.py')
         new_path = Path(editable_lines[-1].split("'")[3])
         if new_path.exists():  # TODO: could remove existence check
             if new_path.name.startswith("__init__"):
-                return new_path.parent.parent
-            return new_path
+                return [new_path.parent.parent]
+            return [new_path]
     elif path.name.startswith("__editable__"):
         # support for how 'setuptools' writes these files:
-        # example line: MAPPING = {'griffe': '/media/data/dev/griffe/src/griffe'}
+        # example line: MAPPING = {'griffe': '/media/data/dev/griffe/src/griffe', 'briffe': '/media/data/dev/griffe/src/briffe'}
         for line in editable_lines:
-            match = _re_mapping_line.match(line)
-            if match:
-                new_path = Path(match.group(1))
-                if new_path.exists():  # TODO: could remove existence check
-                    return new_path.parent
-                break
+            if line.startswith("MAPPING"):
+                paths = ast.literal_eval(line.split(" = ", 1)[1])
+                assert isinstance(paths, dict)
+                return [Path(pt).parent for pt in paths.values()]
```

Comment:
At this point I feel like we should parse the entire file with AST (without any evaluation), and iterate on the module's body to find the `MAPPING` assignment node. Who knows if setuptools won't split the assignment on multiple lines in some cases, or in the future?

It would look like this:

```python
parsed_module = ast.parse(path.read_text())
for node in parsed_module.body:
    if isinstance(node, ast.Assign) and node.targets[0].id == "MAPPING":
        return [Path(constant.value).parent for constant in node.value.values]
```

Reply:
I have changed to full AST parsing, I agree it will be more stable.

A concern I do have at this point is that the change by setuptools or editables or any other package will be bigger than that. 

It is probably not for this MR, and I don't understand the ModuleFinder mechanism enough, but I think it might probably  be augmented by using the sys.meta_hook, and in that case there is no need to specially handle each editable install variant. 

If you prefer, I *think* the code below may replace the `_extend_from_editable_modules`

```python
    def find_package(self, module_name: str) -> Package | NamespacePackage:  # noqa: WPS231
        """Find a package or namespace package.

        Parameters:
            module_name: The module name.

        Raises:
            ModuleNotFoundError: When the module cannot be found.

        Returns:
            A package or namespace package wrapper.
        """
        filepaths = [
            Path(module_name),
            # TODO: handle .py[cod] and .so files?
            Path(f"{module_name}.py"),
        ]
       # Add meta path
        specs_paths = []
        for finder in sys.meta_path:
            spec = finder.find_spec(module_name,None)
            if spec and spec.submodule_search_locations:
                specs_paths.extend(Path(path).parent for path in spec.submodule_search_locations)
        namespace_dirs = []
        for path in self.search_paths + specs_paths:  # noqa: WPS440
            path_contents = self._contents(path)
            if path_contents:
                for choice in filepaths:
                    abs_path = path / choice
                    if abs_path in path_contents:
                        if abs_path.suffix:
                            stubs = abs_path.with_suffix(".pyi")
                            return Package(module_name, abs_path, stubs if stubs.exists() else None)
                        else:
                            init_module = abs_path / "__init__.py"
                            if init_module.exists() and not _is_pkg_style_namespace(init_module):
                                stubs = init_module.with_suffix(".pyi")
                                return Package(module_name, init_module, stubs if stubs.exists() else None)
                            namespace_dirs.append(abs_path)

        if namespace_dirs:
            return NamespacePackage(module_name, namespace_dirs)

        raise ModuleNotFoundError(module_name)
```

Reply:
This is a small change on the griffe code - just added the lines below (which I wrote myself) to the function from griffe, and iterated also on `specs_paths`

```pypthon
        specs_paths = []
        for finder in sys.meta_path:
            spec = finder.find_spec(module_name,None)
            if spec and spec.submodule_search_locations:
                specs_paths.extend(Path(path).parent for path in spec.submodule_search_locations)
```